### ISSUE TITLE
dotnet dev-certs --clean has a misleading description

### ISSUE THREAD

**Issue by NTaylorMullen** [aspnet/DotNetTools#426]
The `dotnet dev-certs --clean` command has a confusing description. It makes it feel like the `--clean` command clears all https dev certificates (including those from VS); which isn't true.

![image](https://user-images.githubusercontent.com/2008729/38513797-2b088c22-3be4-11e8-818e-4872ad377c0b.png)

**Comment by javiercn** [comment]
Are you running behind IIS or Kestrel. The dev-certs tool is only for the certificates kestrel uses

**Comment by NTaylorMullen** [comment]
Behind IIS. If the --clean shouldn't nuke certs behind IIS it should be explicit about what it nukes in its description (it says it nukes every cert in the world associated with localhost https)

**Comment by javiercn** [comment]
It nukes the ones produced for ASP.NET Core, not any other localhost certificate on the machine that you might have.

I’m closing this as by design

**Comment by NTaylorMullen** [comment]
> It nukes the ones produced for ASP.NET Core, not any other localhost certificate on the machine that you might have.
> 
> I’m closing this as by design

@javiercn please fix the description then. Devs will see the VisualStudio experience as a shared experience with the command line. As reference here's the current description:

![image](https://user-images.githubusercontent.com/2008729/38513797-2b088c22-3be4-11e8-818e-4872ad377c0b.png)

**Comment by javiercn** [comment]
I’m failing to understand this. AFAIK the tool never says it cleans localhost certificates nor anything like that. This took only manipulates certificates that are marked with a specific OID. It’s not a general certificate management tool. I think your complain is that the tool is not explicit about the certificates it cleans? But I think your confusion can be caused by a lack of context.

All I’m trying to say is that you should not expect this tool to do anything for environments other than Kestrel.

(A similar thing would happen with fiddler or any other ingress like ngynx or similar)

**Comment by NTaylorMullen** [comment]
> All I’m trying to say is that you should not expect this tool to do anything for environments other than Kestrel.
> I think your complain is that the tool is not explicit about the certificates it cleans? But I think your confusion can be caused by a lack of context.

 `Cleans all HTTPS development certificates from the machine.`

You're right it doesn't say "localhost" but I interpreted `HTTPS development certificates` as one and the same. I misunderstood its capabilities based off of its description, I'm sure others will run into similar misunderstandings. We can't rely on people to have a significant amount of context in this area because this is now a perceived pre-requisite step to run our default templates with a valid cert. 

There's probably a better modification but this would even be an improvement:
`Cleans all HTTPS development certificates created by 'dotnet dev-certs https --trust'.`

**Comment by developer-ly** [comment]
After deleting all localhost (asp.net core) certificates manually, everything started to work normally as described in the documention.

The state I had before resetting everything manually was:
- localhost certificates already existed
- couldn't be cleaned
- a new certificate was generated each time I excuted the command "dotnet dev-certs https" even though a valid one existed

I am not sure, If the problems described below won't reappear tomorrow if i try the same again.
-----------------------------------------------------------------------------------------------------------------
Before the above mentioned (good) results, I wrote the following:

This statement "There was an error trying to clean HTTPS development certificates on this machine." is not really accurate.

here are a few notes from what I experienced using dev-certs:

- I had to navigate to certmgr.msc and manually remove the localhost certs generated by the dev-certs tool (from the personal certificates section)
- The clean-command does not remove any pre-existing certificates
- dev-certs https does not check if there is a valid certificate available for the localhost before generating a new one (not showing a warning either)
- generating certificates multiple times in a row leads to having several localhost-certs which do not get cleaned
